### PR TITLE
[UTOPIA-1005]fix skip to main content nav link

### PR DESCRIPTION
**Pull request by binggaofw**
# Description

This PR includes the following proposed change(s):

-fix skip to main content nav link 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/App.tsx [hunk 1]
```diff
@@ -17,13 +17,31 @@ function App() {
...
+    if (mainContentRef.current) {
+      mainContentRef.current.focus();
+    }
+  };
+
+  useEffect(() => {
+    if (skipToContentRef.current) {
+      skipToContentRef.current.focus(); // Set focus to the main content container when the route changes
+    }
+  }, [pathname]);
```
Comment (kushal-arora-fw) [inline src/frontend/src/App.tsx:32]: I like the idea of pathname as dependency to check for url changes. However, this is bringing the button to focus everytime we come to the screen, which is not an expected behaviour for most of our users.
Can you explore a possibility of somehow reseting tabindex everytime the pathname changes?

Comment (binggaofw) [reply src/frontend/src/App.tsx:32]: do a new update for this issue, could you please check again

**Review by Bottle7 — APPROVED** [review]
Works excellently! Thanks, Bing 🚀

**Review by kushal-arora-fw — APPROVED** [review]
Looks good!

**Pull request merged**